subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuterTest.groovy

```diff
@@ -66,39 +80,73 @@ import static org.gradle.internal.work.AsyncWorkTracker.ProjectLockRetention.REL
 
 class ExecuteActionsTaskExecuterTest extends Specification {
```

Comment:
This test is getting pretty strange now. Maybe we should replace it with a test which checks that the "correct" unit of work is generated without having the unit of work go completely through the execution engine. WDYT?

Reply:
We do have a similar test that is building up from the other side: `IncrementalExecutionIntegrationTest`. I'd like to get rid of this one in favor of what you are proposing, but would like to move the rest of the things though first. And maintain this test meanwhile. WDYT?

Reply:
Sounds good to me.